### PR TITLE
[mobile]Implement addQuicHint() API

### PR DESCRIPTION
**Pull request by RenjieTang**
Commit Message: Implement Envoy Mobile's addQuicHint() API
Additional Description: This allows AlternateProtocolsCache to be prefilled.
Risk Level: Low
Testing: unit tests
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: mobile only

### CODE REVIEW

**Review by alyssawilk — COMMENTED** [review]
Nice addition!   I would like e2e testing of this change.

We could roughly duplicate test/extensions/filters/http/alternate_protocols_cache/filter_integration_test.cc in 
mobile/test/common/integration/client_integration_test.cc - HTTP/3-only upstream so it won't work without the alt-svc cache and would work with.  Alternately I think we could tweak 
mobile/test/java/io/envoyproxy/envoymobile/engine/testing/QuicTestServerTest.java which I think largely used yaml >> builder because there was no way to force HTTP/3 prior to this change.

/wait

File: mobile/library/kotlin/io/envoyproxy/envoymobile/EngineBuilder.kt [hunk 1]
```diff
@@ -731,6 +732,7 @@ open class EngineBuilder(
       enableHttp3,
       http3ConnectionOptions,
       http3ClientConnectionOptions,
+      quicHints,
```
[the remark this answers is not in the document]
Comment (RenjieTang) [reply mobile/library/kotlin/io/envoyproxy/envoymobile/EngineBuilder.kt:748]: I realized we don't even have enableHttp3() in the api doc. Let's do that in a separate PR?

**Comment by RenjieTang** [comment]
> Nice addition! I would like e2e testing of this change.
> 
> We could roughly duplicate test/extensions/filters/http/alternate_protocols_cache/filter_integration_test.cc in mobile/test/common/integration/client_integration_test.cc - HTTP/3-only upstream so it won't work without the alt-svc cache and would work with. Alternately I think we could tweak mobile/test/java/io/envoyproxy/envoymobile/engine/testing/QuicTestServerTest.java which I think largely used yaml >> builder because there was no way to force HTTP/3 prior to this change.
> 
> /wait

**Review by alyssawilk — APPROVED** [review]
SG, fine landing this as-is and having a follow-up with kotlin/swift APIs, docs, and possibly enabling test/java/org/chromium/net/CronetUrlRequestContextTest.java test with addQuicHint

**Pull request merged**